File: app/src/main/java/com/semear/tec/palavrizapp/ui/activities/MainActivity.java
package com.semear.tec.palavrizapp.ui.activities;

import android.arch.lifecycle.ViewModelProviders;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.FrameLayout;

import com.semear.tec.palavrizapp.R;
import com.semear.tec.palavrizapp.models.User;
import com.semear.tec.palavrizapp.ui.fragments.DashboardFragment;
import com.semear.tec.palavrizapp.ui.fragments.PlansFragment;
import com.semear.tec.palavrizapp.viewmodel.DashboardViewModel;

import butterknife.BindView;
import butterknife.ButterKnife;

public class MainActivity extends AppCompatActivity {

    @BindView(R.id.frameContent)
    FrameLayout frameContent;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);

        changeFragment(new DashboardFragment());

    }


    public void changeFragment(Fragment fragment){
        FragmentManager fm = getSupportFragmentManager();
        FragmentTransaction ft = fm.beginTransaction();
        ft.replace(R.id.frameContent, fragment);
        if (!(fragment instanceof DashboardFragment))
            ft.addToBackStack("tag");
        ft.commit();
    }

}

File: app/src/main/java/com/palavrizar/tec/palavrizapp/modules/MainActivity.kt
package com.palavrizar.tec.palavrizapp.modules

import android.arch.lifecycle.Observer
import android.arch.lifecycle.ViewModelProviders
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.Menu
import android.view.MenuItem
import com.palavrizar.tec.palavrizapp.R
import com.palavrizar.tec.palavrizapp.modules.base.BaseActivity
import com.palavrizar.tec.palavrizapp.modules.dashboard.DashboardFragment
import com.palavrizar.tec.palavrizapp.utils.repositories.SessionManager
import com.palavrizar.tec.palavrizapp.modules.classroom.ClassroomActivity
import android.content.Intent
import com.palavrizar.tec.palavrizapp.models.Video
import com.palavrizar.tec.palavrizapp.utils.constants.Constants


class MainActivity : BaseActivity(), DashboardFragment.OnFragmentInteractionListener {

    private var sessionManager: SessionManager? = null
    private var mainViewModel: MainViewModel? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initViewModel()
        sessionManager = SessionManager(applicationContext)
        if (sessionManager?.isUserFirstTime == true){
            startClassroomActivity()
            changeFragment(DashboardFragment(), "Dashboard")
        }else {
            changeFragment(DashboardFragment(), "Dashboard")
        }
    }


    fun setActionBarTitle(text: String) {
        if (supportActionBar != null)
            supportActionBar!!.title = text
    }

    private fun initViewModel() {
        mainViewModel = ViewModelProviders.of(this).get(MainViewModel::class.java)
        mainViewModel?.initViewModel()
    }

    fun changeFragment(fragment: Fragment, fragName: String) {
        val fm = supportFragmentManager
        val ft = fm.beginTransaction()
        ft.replace(R.id.frameContent, fragment, fragName)
        if (fragment !is DashboardFragment) {
            supportActionBar?.setDisplayHomeAsUpEnabled(true)
            ft.addToBackStack("tag")
        }
        ft.commit()
    }


    override fun OnPlansClicked() {

    }

    private fun startClassroomActivity() {
        mainViewModel?.startClassroomActivity(this)
    }

    override fun OnThemesClicked() {

    }
}
